contracts/support/body.go

```diff
@@ -0,0 +1,17 @@
+package support
+
+import "io"
+
+type Body interface {
+	SetFields(fields map[string]any) Body
+	SetField(key string, value any) Body
+	GetField(key string) any
+	SetFiles(files map[string]string) Body
+	SetFile(fieldName, filePath string) Body
+	Build() (BodyContent, error)
+}
+
+type BodyContent interface {
+	Reader() io.Reader
+	ContentType() string
+}
```

Comment:
How about:

```suggestion
type Reader interface {
	io.Reader
	ContentType() string
}
```

Then `Reader` can be passed to `.Post()` which was added in the previous PR directly.

Reply:
This won't work  since they are determining `Content-Length` by calling `Len` method of built-in package.
![image](https://github.com/user-attachments/assets/e7f33c4f-1c33-47b9-a245-c7f4cc187a54)